lib/logstash/inputs/s3.rb

```diff
@@ -143,18 +144,28 @@ def process_files(queue)
     objects = list_new_files
 
     objects.each do |key|
-      @logger.debug("S3 input processing", :bucket => @bucket, :key => key)
-
-      lastmod = @s3bucket.objects[key].last_modified
-
-      process_log(queue, key)
-
-      sincedb.write(lastmod)
+      if stop?
+        break
+      else
+        @logger.debug("S3 input processing", :bucket => @bucket, :key => key)
+        process_log(queue, key)
+      end
     end
   end # def process_files
 
+  public
+  def stop
+    super
```

Comment:
right, things were in flux ;)
the pipeline will call `#do_stop` instead of stop, so we have to remove the call to super.